src/ui/public/kfetch/kfetch.test.ts

```diff
@@ -28,84 +28,458 @@ jest.mock('../metadata', () => ({
+        method: 'GET',
+      });
+    });
+
+    it('should modify request', () => {
+      expect(resp).toEqual({
+        addedByResponseInterceptor: true,
+        foo: 'bar',
+      });
+    });
+  });
+
+  describe('when response interceptor throws', () => {
+    let error: Error;
+    beforeEach(async () => {
+      fetchMock.get('*', { foo: 'bar' });
+      addInterceptor({
+        response: res => {
+          throw new Error('custom response error');
+        },
+      });
+
+      try {
+        await kfetch({ pathname: 'my/path' });
+      } catch (e) {
+        error = e;
+      }
+    });
+
+    it('should return rejected promise', async () => {
+      expect(error.message).toBe('custom response error');
+    });
+  });
+
+  describe('when the first of three response interceptors throws an error', () => {
+    let error: Error;
+    let spy1: jest.Mock;
+    let spy2: jest.Mock;
+    let spy3: jest.Mock;
+
+    beforeEach(async () => {
+      fetchMock.get('*', { foo: 'bar' });
+
+      spy1 = jest.fn(e => {
+        throw new Error('my custom error');
       });
+
+      spy2 = jest.fn();
+      spy3 = jest.fn();
+
+      addInterceptor({ response: spy1 });
+      addInterceptor({ response: spy2 });
+      addInterceptor({ response: spy3 });
+
+      try {
+        await kfetch({ pathname: 'my/path' });
+      } catch (e) {
+        error = e;
+      }
+    });
+
+    it('should call the first interceptor', () => {
+      expect(spy1.mock.calls[0][0]).toEqual({ foo: 'bar' });
+    });
+
+    it('should reject with the error from the first interceptor', () => {
+      expect(error.message).toBe('my custom error');
+    });
+
+    it('should not call subsequent interceptors', () => {
+      expect(spy2).not.toHaveBeenCalled();
+      expect(spy3).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('when all response interceptors resolve', async () => {
+    let resp: any;
+    let spy1: jest.Mock;
+    let spy2: jest.Mock;
+    let spy3: jest.Mock;
+
+    beforeEach(async () => {
+      fetchMock.get('*', { foo: 'bar' });
+
+      spy1 = jest.fn(res => ({ ...res, spy1: true }));
+      spy2 = jest.fn(res => ({ ...res, spy2: true }));
+      spy3 = jest.fn(res => ({ ...res, spy3: true }));
+      addInterceptor({ response: spy1 });
+      addInterceptor({ response: spy2 });
+      addInterceptor({ response: spy3 });
+      resp = await kfetch({ pathname: 'my/path' });
+    });
+
+    it('should return the value of the last interceptor', () => {
+      expect(resp).toEqual({ foo: 'bar', spy1: true, spy2: true, spy3: true });
+    });
+
+    it('should call each interceptors with the value of the preceding interceptor', () => {
+      expect(spy1.mock.calls[0][0]).toEqual({ foo: 'bar' });
+      expect(spy2.mock.calls[0][0]).toEqual({ foo: 'bar', spy1: true });
+      expect(spy3.mock.calls[0][0]).toEqual({ foo: 'bar', spy1: true, spy2: true });
     });
   });
 });
+
+function mockInterceptorCalls(interceptors: Interceptor[]) {
```

Comment:
This helper makes it easier to track the calls made to interceptors (intercepting interceptors...) but might be an over-abstraction. Let me know what you think.